library/core/src/iter/adapters/fuse.rs

```diff
@@ -24,6 +24,10 @@ impl<I> Fuse<I> {
     pub(in crate::iter) fn new(iter: I) -> Fuse<I> {
         Fuse { iter: Some(iter) }
     }
+
+    pub(crate) fn into_inner(self) -> Option<I> {
```

Comment:
Inline or it inlines itself good enough?

Reply:
Well, it worked in the microbenchmarks at least. With mir inlining, the cross-crate auto-inlining which was recently added and std being compiled with 1CGU that's probably enough.
If perf shows regressions I'll reconsider.